KubeArmor/core/kubeUpdate.go

```diff
@@ -2382,12 +2393,103 @@ func (dm *KubeArmorDaemon) removeBackUpPolicy(name string) {
 // == Default Posture == //
 // ===================== //
 
-func validateDefaultPosture(key string, ns *corev1.Namespace, defaultPosture string) string {
+func (dm *KubeArmorDaemon) updatEndpointsWithCM(cm *corev1.ConfigMap, action string) {
+	dm.EndPointsLock.Lock()
+	defer dm.EndPointsLock.Unlock()
+
+	dm.DefaultPosturesLock.Lock()
+	defer dm.DefaultPosturesLock.Unlock()
+
+	// get all namespaces
+	nsList, err := K8s.K8sClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		kg.Err("unable to fetch requested namespace")
+	}
+
+	// for each namespace if needed change endpoint depfault posture
+	for _, ns := range nsList.Items {
+		ns := ns
+		fp, fa := validateDefaultPosture("kubearmor-file-posture", &ns, cm.Data[cfg.ConfigDefaultFilePosture])
+		np, na := validateDefaultPosture("kubearmor-network-posture", &ns, cm.Data[cfg.ConfigDefaultNetworkPosture])
+		cp, ca := validateDefaultPosture("kubearmor-capabilities-posture", &ns, cm.Data[cfg.ConfigDefaultCapabilitiesPosture])
+		annotated := fa || na || ca      // if namespace is annotated for atleast one posture
+		fullyannotated := fa && na && ca // if namespace is fully annotated
+		posture := tp.DefaultPosture{
+			FileAction:         fp,
+			NetworkAction:      np,
+			CapabilitiesAction: cp,
+		}
+		dm.Logger.Printf("namespace %v is fullyannotated %v", ns.Name, fullyannotated)
```

Comment:
This should be in an if statement, as at this point there is no check if the ns is fully annotated

Reply:
these are just part of over-commenting for debug/review purpose, i was intended to remove before merge.